packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts

```diff
@@ -46,100 +42,81 @@ export const inviteMemberHandler = async ({ ctx, input }: InviteMemberOptions) =
 
   const team = await getTeamOrThrow(input.teamId, input.isOrg);
   const { autoAcceptEmailDomain, orgVerified } = getIsOrgVerified(input.isOrg, team);
-
-  const translation = await getTranslation(input.language ?? "en", "common");
-
   const emailsToInvite = await getEmailsToInvite(input.usernameOrEmail);
-
-  for (const usernameOrEmail of emailsToInvite) {
-    const connectionInfo = getOrgConnectionInfo({
-      orgVerified,
-      orgAutoAcceptDomain: autoAcceptEmailDomain,
-      usersEmail: usernameOrEmail,
-      team,
-      isOrg: input.isOrg,
-    });
-    const invitee = await getUserToInviteOrThrowIfExists({
-      usernameOrEmail,
-      teamId: input.teamId,
-      isOrg: input.isOrg,
+  const orgConnectInfoByEmail = emailsToInvite.reduce((acc, email) => {
+    return {
+      ...acc,
+      [email]: getOrgConnectionInfo({
+        orgVerified,
+        orgAutoAcceptDomain: autoAcceptEmailDomain,
+        usersEmail: email,
+        team,
+        isOrg: input.isOrg,
+      }),
+    };
+  }, {} as Record<string, ReturnType<typeof getOrgConnectionInfo>>);
+  const existingUsersWithMembersips = await getUsersToInvite({
+    usernameOrEmail: emailsToInvite,
+    isInvitedToOrg: input.isOrg,
+    team,
+  });
+  const existingUsersEmails = existingUsersWithMembersips.map((user) => user.email);
+  const newUsersEmails = emailsToInvite.filter((email) => !existingUsersEmails.includes(email));
+  // deal with users to create and invite to team/org
+  if (newUsersEmails.length) {
+    await createNewUsersConnectToOrgIfExists({
+      usernamesOrEmails: newUsersEmails,
+      input,
+      connectionInfoMap: orgConnectInfoByEmail,
+      autoAcceptEmailDomain,
+      parentId: team.parentId,
     });
-
-    if (!invitee) {
-      checkInputEmailIsValid(usernameOrEmail);
-
-      // valid email given, create User and add to team
-      await createNewUserConnectToOrgIfExists({
+    const sendVerifEmailsPromises = newUsersEmails.map((usernameOrEmail) => {
+      return sendVerificationEmail({
         usernameOrEmail,
+        team,
+        translation,
+        ctx,
         input,
-        connectionInfo,
-        autoAcceptEmailDomain,
-        parentId: team.parentId,
+        connectionInfo: orgConnectInfoByEmail[usernameOrEmail],
       });
+    });
+    sendEmails(sendVerifEmailsPromises);
```

Comment:
@ThyMinimalDev shouldn't sendEmails be awaited? Are we running a risk of a race condition by not doing so?